rdkit/Chem/AllChem.py

```diff
@@ -464,47 +465,170 @@ def AssignBondOrdersFromTemplate(refmol, mol):
   mol2 = rdchem.Mol(mol)
   # do the molecules match already?
   matching = mol2.GetSubstructMatch(refmol2)
-  if not matching: # no, they don't match
+  if not matching or mol2.GetNumBonds() != refmol2.GetNumBonds(): # no, they don't match, or some bonds are bad
     # check if bonds of mol are SINGLE
     for b in mol2.GetBonds():
-      if b.GetBondType() != BondType.SINGLE:
-        b.SetBondType(BondType.SINGLE)
-        b.SetIsAromatic(False)
+      b.SetBondType(BondType.SINGLE)
+      b.SetIsAromatic(False)
     # set the bonds of mol to SINGLE
     for b in refmol2.GetBonds():
       b.SetBondType(BondType.SINGLE)
       b.SetIsAromatic(False)
     # set atom charges to zero;
     for a in refmol2.GetAtoms():
       a.SetFormalCharge(0)
+      a.SetIsAromatic(False)
     for a in mol2.GetAtoms():
       a.SetFormalCharge(0)
-      
+      a.SetIsAromatic(False)
+
     matching = mol2.GetSubstructMatches(refmol2, uniquify=False)
```

Comment:
I thought that there was a good reason you've used it previously, so I stick to using it. I'll test whether it causes problems and get back to you.